src/main/java/org/circle8/dto/ResiduoDto.java

```diff
@@ -1,48 +1,91 @@
+import org.circle8.entity.User;
 
 public class ResiduoDto {
 	public long id;
-	public long ciudadanoId;
-	public ZonedDateTime fechaRetiro;
 	public ZonedDateTime fechaCreacion;
-	public PuntoResiduoDto punto;
-	public TipoResiduoDto tipo;
+	public ZonedDateTime fechaRetiro;	
+	public ZonedDateTime fechaLimiteRetiro;
+	public String descripcion;
+	public Long ciudadanoId;
+	public Long puntoResiduoId;
+	public PuntoResiduoDto puntoResiduo;
+	public Long tipoResiduoId;
+	public TipoResiduoDto tipoResiduo;
+	public Long recorridoId;
 	public RecorridoDto recorrido;
+	public Long transaccionId;
 	public TransaccionDto transaccion;
-
+	
+	public static ResiduoDto from(PostResiduoRequest req) {
+		var r = new ResiduoDto();
+		r.fechaLimiteRetiro = req.fechaLimite;
+		r.descripcion = req.descripcion;
+		r.ciudadanoId = req.ciudadanoId;
+		r.puntoResiduoId = req.puntoResiduoId;
+		var pr = PuntoResiduo.builder()
+				.id(req.puntoResiduoId)
+				.ciudadano(User.builder().id(req.ciudadanoId).build())
+				.residuos(Collections.emptyList())
+				.build();
+		r.puntoResiduo = PuntoResiduoDto.from(pr);
+		r.tipoResiduoId = req.tipoResiduoId;
+		r.tipoResiduo = TipoResiduoDto.from(TipoResiduo.builder().id(req.tipoResiduoId).build());
+		return r;
+	}
+	
 	public static ResiduoDto from(Residuo entity) {
 		var r = new ResiduoDto();
 		r.id = entity.id;
 		r.ciudadanoId = entity.ciudadanoId;
-		r.fechaRetiro = entity.fechaRetiro;
 		r.fechaCreacion = entity.fechaCreacion;
-		r.tipo = TipoResiduoDto.from(entity.tipo);
-		r.punto = PuntoResiduoDto.from(entity.punto);
+		r.fechaRetiro = entity.fechaRetiro;
+		r.fechaLimiteRetiro = entity.fechaLimiteRetiro;
+		r.descripcion = entity.descripcion;
+		r.puntoResiduo = PuntoResiduoDto.from(entity.puntoResiduo);
+		r.tipoResiduoId = r.tipoResiduoId;
+		r.tipoResiduo = TipoResiduoDto.from(entity.tipoResiduo);
 		r.recorrido = RecorridoDto.from(entity.recorrido);
 		r.transaccion = TransaccionDto.from(entity.transaccion);
 		return r;
 	}
-
+	
 	public ResiduoResponse toResponse() {
 		return new ResiduoResponse(
-			id,
-			fechaRetiro,
-			fechaCreacion,
-			"/ciudadano/"+ciudadanoId+"/punto_residuo/"+punto.id,
-			punto.id,
-			punto.toResponse(),
-			tipo.toResponse(),
-			recorrido != null ? "/recorrido/" + recorrido.id : null,
-			recorrido != null ? recorrido.id : null,
-			recorrido != null ? recorrido.toResponse() : null,
-			transaccion != null ? "/transaccion/" + transaccion.id : null,
-			transaccion != null ? transaccion.id : null,
-			transaccion != null ? transaccion.toResponse() : null
-		);
+				this.id,
+				this.fechaRetiro,
+				this.fechaCreacion,
+				this.fechaLimiteRetiro,
+				this.descripcion,
+				"/ciudadano/"+this.ciudadanoId+"/punto_residuo/"+this.puntoResiduo.id,
+				this.puntoResiduo.id,
+				this.puntoResiduo.toResponse(),
+				this.tipoResiduo.toResponse(),
+				this.recorrido != null ? "/recorrido/"+this.recorrido.id : null,
+				this.recorridoId != null ? this.recorrido.id : null,
```

Comment:
Esto deberia ser `this.recorridoId` y nada mas
Lo que se esta haciendo es `a != null ? b.id : null` y si `b` es null, rompe todo

Reply:
Como decia el otro comentario, se saco recorrido por lo que aca directamente se usa this.recorridoId